public/resources/remix/index.js

```diff
@@ -6,122 +6,136 @@
+    xmlhttp.send(data);
   }
 
-  function injectStyleSheets($, metadata) {
-    var stylesheets =
-      "<link href=\"" + metadata.host + "/resources/remix/clean-slate.min.css\" rel=\"stylesheet\">\n" +
-      "<link href=\"" + metadata.host + "/resources/remix/style.css\" rel=\"stylesheet\">\n";
-    $("head").append(stylesheets);
+  function injectStyleSheets(metadata) {
+    var stylesheet = document.createElement("link");
+    stylesheet.rel = "stylesheet";
+    stylesheet.href = metadata.host + "/resources/remix/clean-slate.min.css";
+    document.head.appendChild(stylesheet);
+
+    stylesheet = document.createElement("link");
+    stylesheet.rel = "stylesheet";
+    stylesheet.href = metadata.host + "/resources/remix/style.css";
+    document.head.appendChild(stylesheet);
+  }
+
+  function grep(items, callback) {
+    var filtered = [], len = items.length, i = 0;
+    for (i; i < len; i++) {
+      var item = items[i];
+      var cond = callback(item);
+      if (cond) {
+        filtered.push(item);
+      }
+    }
+    return filtered;
   }
 
-  function getMetadata($) {
+  function getMetadata() {
     var metadata = {};
-    var metaTags = $("meta");
+    var metaTags = document.getElementsByTagName("meta");
 
-    $.grep(metaTags, function(elem, index) {
-      return /^data-remix-.+/.test($(metaTags[index]).attr("name"));
+    grep(metaTags, function(metaTag) {
+      return /^data-remix-.+/.test((metaTag).getAttribute("name"));
     })
     .forEach(function(metaTag) {
-      metaTag = $(metaTag);
-      var key = metaTag.attr("name").replace(/^data-remix-/, "");
+      var key = metaTag.getAttribute("name").replace(/^data-remix-/, "");
+
       // We see if a value for the current attribute already exists.
       // If it does, that means that we already encountered a meta tag
       // with that name which will refer to a more recent value than the
       // current one (we add the most recent values above all other values in
       // the <head> tag's contents).
-      metadata[key] = metadata[key] || metaTag.attr("content");
+      metadata[key] = metadata[key] || metaTag.getAttribute("content");
     });
 
     return metadata;
   }
 
   function run() {
-    var $$ = $.noConflict(true);
-    $$(document).ready(function($) {
-      var metadata = getMetadata($);
-      injectAnalytics($);
-      injectStyleSheets($, metadata);
-      injectDetailsBar($, metadata, setupBar);
+    document.addEventListener("DOMContentLoaded", function() {
+      var metadata = getMetadata();
+      injectAnalytics();
+      injectStyleSheets(metadata);
+      injectDetailsBar(metadata, setupBar);
     });
   }
 
-  (function(doc, script) {
-    script = doc.createElement("script");
-    script.type = "text/javascript";
-    script.onload = run;
-    script.src = "https://ajax.googleapis.com/ajax/libs/jquery/2.1.4/jquery.min.js";
-    doc.getElementsByTagName("head")[0].appendChild(script);
-  }(document));
+  run();  
```

Comment:
if this function doesn't need to be called anywhere else, and it is immediately invoked, then we don't need this function. We just need the code inside of it to immediately run:

```
function dothis() {
  console.log("test");
}
dothis();
```
executes identically to:
```
console.log("test");
```
but without the function allocation and call overhead.